src/object.d

```diff
@@ -3241,6 +3241,40 @@ template RTInfo(T)
     enum RTInfo = null;
 }
 
+bool __equals(T1, T2)(T1[] lhs, T2[] rhs)
+{
+    if (lhs.length != rhs.length)
+        return 0;
+
+    import core.internal.traits : Unqual;
+    alias U1 = Unqual!T1;
+    alias U2 = Unqual!T2;
+    static assert(is(U1 == U2), "Internal error.");
```

Comment:
No need to define names because we're never using them, `static assert(is(Unqual!T1 == Unqual!T2), "Internal error.");` should suffice